catalog/spatial/registry/registry-publication-update-handler/src/main/java/org/codice/ddf/registry/publication/RegistryPublicationHandler.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.publication;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.codice.ddf.registry.common.RegistryConstants;
+import org.codice.ddf.registry.common.metacard.RegistryObjectMetacardType;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminException;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+
+public class RegistryPublicationHandler implements EventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryPublicationHandler.class);
+
+    private static final String METACARD_PROPERTY = "ddf.catalog.event.metacard";
+
+    private FederationAdminService adminService;
+
+    private ExecutorService executor;
+
+    private int threadPoolSize = 1;
+
+    public RegistryPublicationHandler(FederationAdminService adminService) {
+        this.adminService = adminService;
+    }
+
+    public void init() {
+        if (executor == null) {
+            executor = Executors.newFixedThreadPool(threadPoolSize);
+        }
+    }
+
+    public void setExecutorService(ExecutorService service) {
+        executor = service;
+    }
+
+    public void destroy() {
+        executor.shutdown();
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        Metacard mcard = (Metacard) event.getProperty(METACARD_PROPERTY);
+        if (mcard == null) {
```

Comment:
We could also combine the two checks instead of having two if statements:
```
if ((mcard == null) || !mcard.getTags().contains(RegistryConstants.REGISTRY_TAG)) {
    return;
}
```